packages/transaction-controller/src/TransactionController.ts

```diff
@@ -607,8 +636,48 @@ export class TransactionController extends BaseControllerV1<
     });
 
     this.onBootCleanup();
+    this.#initTrackingMap();
   }
 
+  #refreshTrackingMap = () => {
+    const networkClients = this.getNetworkClientRegistry();
+    const networkClientIds = Object.keys(networkClients);
+
+    const existingNetworkClientIds = Array.from(this.trackingMap.keys());
+
+    // Remove tracking for NetworkClientIds that no longer exist
+    const networkClientIdsToRemove = existingNetworkClientIds.filter(
+      (id) => !networkClientIds.includes(id),
+    );
+    networkClientIdsToRemove.forEach((id) => {
+      this.stopTrackingByNetworkClientId(id);
+    });
+
+    if (networkClientIdsToRemove.length > 0) {
+      this.hub.emit('tracking-map-remove', networkClientIdsToRemove);
+    }
+
+    // Start tracking new NetworkClientIds from the registry
+    const networkClientIdsToAdd = networkClientIds.filter(
+      (id) => !existingNetworkClientIds.includes(id),
+    );
+    networkClientIdsToAdd.forEach((id) => {
+      this.startTrackingByNetworkClientId(id);
+    });
+
+    if (networkClientIdsToAdd.length > 0) {
+      this.hub.emit('tracking-map-add', networkClientIdsToAdd);
+    }
+  };
+
+  #initTrackingMap = () => {
+    const networkClients = this.getNetworkClientRegistry();
+    Object.keys(networkClients).map((id) =>
+      this.startTrackingByNetworkClientId(id),
+    );
+    this.hub.emit('tracking-map-init', Object.keys(networkClients));
```

Comment:
[nit]
```suggestion
    this.hub.emit('tracking-map-refreshed', Object.keys(networkClients));
```

Reply:
this only gets called on init, `refreshed` is the name of the other method

Reply:
oh not sure how I misread that